### PR TITLE
Allow beta releases

### PR DESCRIPTION
**Pull request by Tankenstein**
Parses beta/alpha/rc versions and runs pre-releases.

### CODE REVIEW

File: bin/parser/index.js [hunk 1]
```diff
@@ -1,10 +1,14 @@
 function parseChangelogItem(item) {
-  const regex =  /#\s?(v\d+\.\d+\.?\d*)\n##\s?(.*)([\s\S]*)/g;
+  const regex = /#\s?(v\d+\.\d+\.?\d*(-(beta|alpha|rc)\.\d+)?)\n##\s?(.*)([\s\S]*)/g;
   const match = regex.exec(item);
 
-  const [tagName, releaseTitle, description] = match.slice(1,4);
+  const tagName = match[1];
```
Comment (oliverviljamaa) [inline bin/parser/index.js]: How about:
```js
const { 1: tagName, 2: prereleasePart, 4: releaseTitle, 5: description } = match;
```

File: bin/publish-github-release/index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ class GithubReleaseClient {
     this.branch = branch || 'master';
   }
 
-  publishRelease(version, releaseTitle, releaseDescription) {
+  publishRelease(version, releaseTitle, releaseDescription, preRelease) {
```
Comment (oliverviljamaa) [inline bin/publish-github-release/index.js]: Should it be `prerelease` as it is not two words?

Comment (Tankenstein) [reply bin/publish-github-release/index.js]: aight

File: bin/publish-github-release/index.js [hunk 3]
```diff
@@ -62,13 +63,16 @@ function formGithubReleaseResource({ version, title, description, branch }) {
   if (description) {
     releaseResource.body = description;
   }
+  if (preRelease !== undefined) {
```
Comment (oliverviljamaa) [inline bin/publish-github-release/index.js]: Why wouldn't a falsy check suffice?

Comment (Tankenstein) [reply bin/publish-github-release/index.js]: i wanted to be able to pass "false" if you wanted to. Doesn't make a difference though.

**Review by oliverviljamaa — APPROVED** [review]

**Review by adrienDog — APPROVED** [review]

**Pull request merged**